corehq/apps/app_manager/models.py

```diff
@@ -649,25 +649,27 @@ def generate_id(cls):
         return hex(random.getrandbits(160))[2:-1]
 
     @classmethod
-    def get_form(cls, form_unique_id, and_app=False):
-        try:
-            d = get_db().view(
-                'app_manager/xforms_index',
-                key=form_unique_id
-            ).one()
-        except MultipleResultsFound as e:
-            raise XFormIdNotUnique(
-                "xform id '%s' not unique: %s" % (form_unique_id, e)
-            )
-        if d:
-            d = d['value']
-        else:
-            raise ResourceNotFound()
-        # unpack the dict into variables app_id, module_id, form_id
-        app_id, unique_id = [d[key] for key in ('app_id', 'unique_id')]
+    def get_form(cls, form_unique_id, and_app=True):
+        @quickcache(['form_id'], timeout=30 * 60)
+        def get_app_id(form_id):
+            try:
+                d = get_db().view(
+                    'app_manager/xforms_index',
+                    key=form_id
+                ).one()
```

Comment:
It looks like we can actually get the app in a single query instead of getting the id first, then going back for the app. I believe if you you use `include_docs=True` in this view function it will return the app document. I'm not sure if cacheing the app would have the same issue that cacheing a form does though.